src/rgw/rgw_rados.cc

```diff
@@ -1302,6 +1302,10 @@ int RGWPeriod::update()
     return ret;
   }
 
+  // clear zone short ids of removed zones. period_map.update() will add the
+  // remaining zones back
+  period_map.short_zone_ids.clear();
```

Comment:
> zones that were already assigned an id after collision will have new id assigned to them

sorry, i'm not quite sure what you mean here

if you get into a situation where you have multiple zone uuids that hash to the same short id, your `period update --commit` will error out with `New zone <name> (<id>) generates the same short_zone_id <short_id> as existing zone id <id>`

to resolve that, you would have to delete and recreate the new zone, which would generate a new uuid. then the next `period update --commit` would calculate a different short id for it, so there would be no collision and `RGWPeriodMap::update()` would end up with the right `short_zone_ids`

i don't see why we'd need to detect removed zones and handle them differently